src/components/Table/sgds-table.ts

```diff
@@ -21,86 +44,148 @@ export class SgdsTable extends SgdsElement {
 
   /**
-   * Populate data cells using Arrays
+   * Populate data cells using Arrays. For custom rendering of cell use <code>ICellItem</code>.
    */
-  @property({ type: Array }) tableData: Array<(string | number)[]> = [];
+  @property({ type: Array }) tableData: unknown[] = [];
 
   /**
    * Defines the placement of headers in the table (horizontal, vertical, or both)
    */
   @property({ type: String, reflect: true }) headerPosition: HeaderPosition = "horizontal";
 
   /** @internal */
-  @state() originalTableData: Array<(string | number)[]> = [];
+  @state() originalTableData: unknown[] = [];
 
   connectedCallback() {
     super.connectedCallback();
     this.originalTableData = [...this.tableData];
   }
 
-  private _renderTable() {
-    if (this.headerPosition === "horizontal") {
-      return html`
-        <thead>
-          <tr>
-            ${this.rowHeader.map((header: string, index: number) => html` <th>${header}</th> `)}
-          </tr>
-        </thead>
-        <tbody>
-          ${this.tableData.map(
-            row => html`
-              <tr>
-                ${row.map((cell: string) => html`<td>${cell}</td>`)}
-              </tr>
-            `
-          )}
-        </tbody>
-      `;
-    }
+  private _handleClick(e: Event, cell: ICellItem, row: unknown) {
+    const eventName = `sgds-table-click`;
+    const customClickEvent = this.emit(eventName, {
+      detail: {
+        target: cell,
+        row
+      },
+      cancelable: true
+    });
+
+    if (customClickEvent.defaultPrevented) return;
+    e.preventDefault();
+  }
 
-    if (this.headerPosition === "both") {
-      return html`
-        <thead>
-          <tr>
-            <th></th>
-            ${this.rowHeader.map((header: string, index: number) => html` <th>${header}</th> `)}
-          </tr>
-        </thead>
-        <tbody>
-          ${this.tableData.map(
-            (row, index) => html`
-              <tr>
-                <th>${this.columnHeader[index]}</th>
-                ${row.map((cell: string) => html`<td>${cell}</td>`)}
-              </tr>
-            `
-          )}
-        </tbody>
+  private _mapElementType(cell: ICellItem, row: unknown) {
+    const val = cell?.value;
+
+    switch (cell?.type) {
+      case "link":
+        return html`
+          <sgds-link slot="link" id="${cell.id}">
+            <a href="${cell.href ?? "#"}" @click=${(e: Event) => this._handleClick(e, cell, row)}>${val}</a>
+          </sgds-link>
+        `;
+
+      case "button":
+        return html`
+          <sgds-button
+            id="${cell.id}"
+            .fullWidth=${cell.fullWidth}
+            variant=${cell.variant}
+            size=${cell.size ?? "sm"}
+            .active=${cell.active}
+            .disabled=${cell.disabled}
+            href=${cell.href}
+            target=${cell.target}
+            download=${cell.download}
+            ariaLabel=${cell.ariaLabel}
+            @click=${(e: Event) => this._handleClick(e, cell, row)}
```

Comment:
If the same sgds-table-click event to forward down to icon-button, button etc. then if user uses a few diff buttons, a single click will trigger all actions for all button.

Reply:
i think we need to re-discuss how to allow users to pass any button/ badge components themselves through slots and decouple with our table logic